### PR TITLE
Fix varchar IS DISTINCT FROM coerce

### PR DESCRIPTION
**Pull request by miniway**
varchar `IS DISTINCT FROM` varchar(n) causes a coerce exception at the 0.2x versions of Presto. 
```
presto:current> select * from "com.facebook.presto.execution.executor:name=multilevelsplitqueue" where node is distinct from 'x';
Query 20181015_132239_00000_a2dm9 failed: Mismatched types: varchar(1) vs varchar
```

### CODE REVIEW

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java [hunk 1]
```diff
@@ -190,6 +190,12 @@ public void testDuplicatedRowCreateTable()
                 "line 1:49: Column name 'orderkey' specified more than once");
     }
 
+    @Test
+    public void testVarcharIsDistinctFrom()
```
Comment (martint) [inline presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: This should go in AbstractTestQueries, not in the connector integration tests.

File: presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java [hunk 2]
```diff
@@ -238,13 +238,13 @@ public static long hashCode(@SqlType("varchar(x)") Slice value)
...
     }
 
-    @LiteralParameters({"x", "y"})
+    @LiteralParameters("x")
     @ScalarOperator(IS_DISTINCT_FROM)
     @SqlType(StandardTypes.BOOLEAN)
     public static boolean isDistinctFrom(
             @SqlType("varchar(x)") Slice left,
             @IsNull boolean leftNull,
-            @SqlType("varchar(y)") Slice right,
```
Comment (findepi) [inline presto-main/src/main/java/com/facebook/presto/type/VarcharOperators.java]: The original function signature looked like a legit attempt to avoid coercion. What was wrong with that?

File: presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java (hunk 1 above)
Comment (miniway) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: I was not able to simply create a non-varchar(n) table on AbstractTestQueries. `CAST` also didn't help to reproduce the issue as `DomainTranslator.visitComparisonExpression` has the own coercing. Let me try again.

Comment (miniway) [reply presto-tests/src/main/java/com/facebook/presto/tests/AbstractTestIntegrationSmokeTest.java]: I could use tables under information_schema
